### PR TITLE
test(developer): Add tests for automatic versioning of notany() with context()

### PR DESCRIPTION
**Pull request by mcdurdin**
Adds a unit test to verify that automatic versioning of notany() in conjunction with context() is correctly managed. Removes compiler error that could never be generated (replace with safety-check exception).

Fixes: #11959

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-kmn/src/compiler/kmn-compiler-messages.ts [hunk 1]
```diff
@@ -484,10 +484,10 @@ export class KmnCompilerMessages {
     `Invalid key identifier "${def(o.keyId)}"`);
 
   static ERROR_90FeatureOnlyLayoutFile                        = SevError | 0x05C;
-  static Error_90FeatureOnlyLayoutFile                        = () => m(this.ERROR_90FeatureOnlyLayoutFile, `Touch layout file reference requires store(version) '9.0'or higher`);
+  static Error_90FeatureOnlyLayoutFile                        = () => m(this.ERROR_90FeatureOnlyLayoutFile, `Touch layout file reference requires store(version) '9.0' or higher`);
```
Comment (mcdurdin) [inline developer/src/kmc-kmn/src/compiler/kmn-compiler-messages.ts:487]: noticed this minor typo while resolving this

**Review by jahorton — APPROVED** [review]
So... there's no use case for `notany` without a right-hand `context`?   Granted, I suppose that the `notany` character would get deleted if not referenced via `context`.

If there were such a use case, I'm not confident that `Error_NotAnyRequiresVersion14` is inaccessible.

Outside of that concern, LGTM.  It's also a low-priority enough concern to me that I'll go ahead and tick that "approve" box anyway.

**Comment by mcdurdin** [comment]
> If there were such a use case, I'm not confident that `Error_NotAnyRequiresVersion14` is inaccessible.

While it is possible (albeit not useful) to use `notany()` without a corresponding `context()`, I don't think it can result in this error. The error is generated (when required) by the compiler code that handles the `context()` statement on the RHS of the rule. 

The function in which it is tested is the [`ContextChar`](https://github.com/keymanapp/keyman/blob/56aafe5b423f4b337711c20f6c9a8989cdcbfb62/developer/src/kmc-kmn/src/kmw-compiler/javascript-strings.ts#L660) function, which is a closure within the larger, hideous [`JavaScript_OutputString`](https://github.com/keymanapp/keyman/blob/56aafe5b423f4b337711c20f6c9a8989cdcbfb62/developer/src/kmc-kmn/src/kmw-compiler/javascript-strings.ts#L634) function.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.73-alpha](https://keyman.com/downloads/releases/alpha/18.0.73)

**Pull request merged**